pkg/server/container_execsync.go

```diff
@@ -70,6 +73,40 @@ type execOptions struct {
 	timeout time.Duration
 }
 
+func (c *criService) getContainer(id string) (*container.Container, error) {
+	// Get container from our container store.
+	cntr, err := c.containerStore.Get(id)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to find container %q in store", id)
+	}
+	return &cntr, nil
+}
+
+func (c *criService) getSandbox(id string) (*sandbox.Sandbox, error) {
+	// Get sandbox from our sandbox store.
+	sb, err := c.sandboxStore.Get(id)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to find sandbox %q in store", id)
+	}
+	return &sb, nil
+}
+
+func checkContainerState(cntr *container.Container) error {
+	state := cntr.Status.Get().State()
+	if state != runtime.ContainerState_CONTAINER_RUNNING {
+		return errors.Errorf("container is in %s state", criContainerStateToString(state))
+	}
+	return nil
+}
+
+func checkSandboxState(sb *sandbox.Sandbox) error {
+	state := sb.Status.Get().State
+	if state != sandbox.StateReady {
+		return errors.Errorf("sandbox is in %s state", fmt.Sprint(state))
+	}
+	return nil
+}
+
```

Comment:
I prefer extract the common part (unchanged part in this PR) out to be a helper function and use it in `execInContainer` and `portforward`. :)

The function is called `execInContainer`, it feels weird to also handle sandbox here.

Reply:
Done.